### PR TITLE
Update infping to use InfluxDB go client v2

### PR DESCRIPTION
**Pull request by liv3d**
Changed infping to use the v2 InfluxDB go client, this fixes an error
case where we're getting 204 (accepted) responses when Influx was in
turn throwing the data away.

Added a gitignore file which ignores the local build of things

### CODE REVIEW

**Comment by darkliquid** [comment]
needs running through gofmt as the indentation is often wrong (should be tabs, but in many cases is 2 spaces)

File: infping.go [hunk 1]
```diff
@@ -4,156 +4,149 @@
...
+			"avg":  avg,
+			"max":  max,
+		}
+	} else {
+		fields = map[string]interface{}{
+			"loss": loss,
+		}
+	}
+
+	bp, _ := client.NewBatchPoints(client.BatchPointsConfig{
```
Comment (darkliquid) [inline infping.go:103]: This batch should be setup outside of this function and passed in, as the writePints function is called within the scan loop in readPoints. At present, the behaviour is such that it creates a separate batch for every point it is sending, while I imagine you actually want to send a batch for every run of the readPoints loop.

File: infping.go [hunk 2]
```diff
@@ -4,156 +4,149 @@
...
+	cmd := exec.Command("/usr/bin/fping", args...)
+	stdout, err := cmd.StdoutPipe()
+	herr(err)
+	stderr, err := cmd.StderrPipe()
+	herr(err)
+	cmd.Start()
+	perr(err)
+
+	buff := bufio.NewScanner(stderr)
+	for buff.Scan() {
```
Comment (darkliquid) [inline infping.go:53]: Outside this loop is where you want your batch setup, ideally, then pass it into the writePoints method.

File: infping.go [hunk 3]
```diff
@@ -4,156 +4,149 @@
...
+	})
+
+	pt, err := client.NewPoint("ping", tags, fields, time.Now())
+	if err != nil {
+		log.Fatal("Error: ", err)
+	}
+
+	bp.AddPoint(pt)
+
+	err = clnt.Write(bp)
```
Comment (darkliquid) [inline infping.go:115]: This wants to be moved outside the writePoints func and outside the readPoints loop too, so you are only sending the batch after having collected all the data points.

File: infping.go (hunk 1 above)
Comment (liv3d) [reply infping.go:103]: This is as per the upstream documentation - https://github.com/influxdata/influxdb/tree/master/client which also does a batch point within the function.

File: infping.go (hunk 2 above)
Comment (liv3d) [reply infping.go:53]: This isn't going to be touched - the readpoint function is out of scope for this change (it's from the upstream/initial module)

File: infping.go (hunk 3 above)
Comment (liv3d) [reply infping.go:115]: In theory the batch process should be doing X number of hosts at this point, so the write should be correct here.

**Comment by liv3d** [comment]
I'm going to merge this for now; I'll come back and fix issues with it later.

**Pull request merged**